x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/all/rules_table_utility_bar.test.tsx

```diff
@@ -9,52 +9,72 @@ import React from 'react';
 import { mount } from 'enzyme';
 import { waitFor } from '@testing-library/react';
 
-import { AllRulesUtilityBar } from './utility_bar';
+import { AllRulesUtilityBar } from './rules_table_utility_bar';
 import { TestProviders } from '../../../../../common/mock';
 
+jest.mock('./rules_table/rules_table_context');
+
 describe('AllRules', () => {
   it('renders AllRulesUtilityBar total rules and selected rules', () => {
     const wrapper = mount(
       <TestProviders>
         <AllRulesUtilityBar
           canBulkEdit
           onRefresh={jest.fn()}
-          paginationTotal={4}
+          pagination={{
+            page: 1,
+            perPage: 10,
+            total: 21,
+          }}
           numberSelectedItems={1}
           onGetBulkItemsPopoverContent={jest.fn()}
           isAutoRefreshOn={true}
           onRefreshSwitch={jest.fn()}
-          hasBulkActions
         />
       </TestProviders>
     );
 
-    expect(wrapper.find('[data-test-subj="showingRules"]').at(0).text()).toEqual('Showing 4 rules');
+    expect(wrapper.find('[data-test-subj="showingRules"]').at(0).text()).toEqual(
+      'Showing 1-10 of 21 rules'
+    );
     expect(wrapper.find('[data-test-subj="selectedRules"]').at(0).text()).toEqual(
       'Selected 1 rule'
     );
   });
 
-  it('does not render total selected and bulk actions when "hasBulkActions" is false', () => {
-    const wrapper = mount(
-      <TestProviders>
-        <AllRulesUtilityBar
-          canBulkEdit
-          onRefresh={jest.fn()}
-          paginationTotal={4}
-          numberSelectedItems={1}
-          onGetBulkItemsPopoverContent={jest.fn()}
-          isAutoRefreshOn={true}
-          onRefreshSwitch={jest.fn()}
-          hasBulkActions={false}
-        />
-      </TestProviders>
-    );
-
-    expect(wrapper.find('[data-test-subj="showingRules"]').exists()).toBeFalsy();
-    expect(wrapper.find('[data-test-subj="tableBulkActions"]').exists()).toBeFalsy();
-    expect(wrapper.find('[data-test-subj="showingExceptionLists"]').at(0).text()).toEqual(
-      'Showing 4 lists'
+  describe('renders correct pagination label when', () => {
+    it.each([
```

Comment:
Nit: I usually agitate people not to use `test.each` whenever possible as it has several considerable downsides.

- It makes tests harder to read. People usually read code from top to down. But `test.each` force you to read code from down to top because you need to read first through the test body to understand what values provided in cases mean.
- `test.each` makes tests harder to debug. It becomes not easy to put a breaking point in the test body or isolate a single failing case.
- `test.each` allows developers to create a large number of test cases without any clear difference between them and without any indication of what matters. Like here https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/server/usage/detections/rules/update_usage.test.ts#L192-L225

Reply:
I understand what you mean, I was just taking a look at the [discussion on the same topic you had recently](https://github.com/elastic/kibana/pull/126063#discussion_r927550563).

I would still like to test the cases I mentioned above, but don't want to clutter this test file with tests specific to the label generator function. I think the best to do then is to unit test the `getShowingRulesParams` function instead

Reply:
> I think the best to do then is to unit test the `getShowingRulesParams` function instead.
> And in this file, just assert in a single test case that the pagination label is displayed.

Yeah, that makes sense to me 👍